kube-client/src/error.rs

```diff
@@ -19,20 +21,20 @@ pub enum Error {
 
     /// ConnectionError for when TcpStream fails to connect.
     #[error("ConnectionError: {0}")]
-    Connection(std::io::Error),
+    Connection(#[source] std::io::Error),
```

Comment:
Does this not give us a source error with the source's error msg in the display? Should we not be removing `{0}`'s here as per recommendations discussed in https://github.com/kube-rs/kube-rs/pull/515 ?

Reply:
Ah, I see that's out of scope for this PR but listed in the overall goals. :+1:

Reply:
Yeah, I added a task for that in #688. All of our errors with source does this, so I think we should change all of them in a separate PR.